### DISCUSSION TITLE
Use Cloud Shell in Oracle Cloud

### DISCUSSION THREAD

**Discussion by neitsab** [chadgeary/cloudblock#102] [Ideas]
As mentioned in #101, I used OCI [Cloud Shell](https://docs.cloud.oracle.com/Content/API/Concepts/cloudshellintro.htm) to perform Cloudblock install. I want to discuss the pros and cons of this approach:

* **pros:** no need to install git, terraform, oci-cli... Everything is there already. We may also make use of the already existing authentication (?) 
* **cons:** see [limits](https://docs.oracle.com/iaas/Content/API/Concepts/cloudshellintro.htm#Cloud_Shell_Limitations). To summarize, it is in the end a temporary environment holding precious/sensitive data (private key...), and without connection for several months will get destroyed.

It does offer to upload/download files from the menu, so anything sensitive/precious can be retrieved easily.

I'd say that to avoid installing the ridiculous oci-cli on one's local system, it is kind of worth it!

Any inputs on this topic?

**Comment by chadgeary** [comment]
Overall, this would be an excellent use case for cloudshell. I agree it
would be a better experience compared to local tool installation.

That said, neither cloudshell nor my guide do a good job handling the
terraform state file. If the current guide as written was re-written for
cloudshell and mentioned copying the state somewhere secure I'd be cool
with it.
